netket/graph/special.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence, Union
+import numpy as np
+
+from .lattice import Lattice
+
+
+def Grid(length: Sequence[int], *, pbc: Union[bool, Sequence[bool]] = True) -> Lattice:
```

Comment:
Is this feature-equivalent to the current `Grid`? It doesn't seem like it, I'd prefer not to loose the colored edges option, for example, as part of this PR.

Reply:
Coloured edges aren't implemented in `Lattice` and shouldn't be part of this PR. I can add it to the `changelog`

Reply:
Yes, but since you remove `Grid` this is in principle a regression. If that's the only one we could maybe consider it, since this particular feature only got added during the v3 development.

Reply:
I think the features it gains (namely, everything in `Lattice` and being hooked up with the `symmetry` architecture) compensate for it. It also really bothers me that you can't have coloured/further-neighbour edges in `Lattice`, so this omission is probably temporary too.